File: apps/sdkwork-music-h5/src/services/musicService.ts
import { MusicHomeShelf } from '../types/music';

const API_BASE_URL = import.meta.env.VITE_API_BASE_URL || 'http://127.0.0.1:18080';
const APP_API_PREFIX = import.meta.env.VITE_APP_API_PREFIX || '/app/v3/api';

class MusicService {
  private baseUrl: string;

  constructor() {
    this.baseUrl = `${API_BASE_URL}${APP_API_PREFIX}/music`;
  }

  async getHomeShelves(cursor?: string, limit?: number): Promise<MusicHomeShelf[]> {
    const params = new URLSearchParams();
    if (cursor) params.append('cursor', cursor);
    if (limit) params.append('limit', limit.toString());

    const url = `${this.baseUrl}/home/shelves${params.toString() ? `?${params.toString()}` : ''}`;
    
    const response = await fetch(url, {
      headers: {
        'Authorization': `Bearer ${this.getAccessToken()}`,
        'Access-Token': this.getAccessToken(),
      },
    });

    if (!response.ok) {
      throw new Error(`Failed to fetch home shelves: ${response.statusText}`);
    }

    return response.json();
  }

  private getAccessToken(): string {
    // TODO: Get access token from auth context
    return '';
  }
}

export const musicService = new MusicService();

File: apps/sdkwork-music-flutter-mobile/lib/services/music_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/music.dart';

class MusicService {
  static const String _baseUrl = 'http://127.0.0.1:18080';
  static const String _appApiPrefix = '/app/v3/api';

  Future<List<MusicHomeShelf>> getHomeShelves({
    String? cursor,
    int? limit,
  }) async {
    final queryParams = <String, String>{};
    if (cursor != null) queryParams['cursor'] = cursor;
    if (limit != null) queryParams['limit'] = limit.toString();

    final uri = Uri.parse('$_baseUrl$_appApiPrefix/music/home/shelves')
        .replace(queryParameters: queryParams);

    final response = await http.get(
      uri,
      headers: {
        'Authorization': 'Bearer ${_getAccessToken()}',
        'Access-Token': _getAccessToken(),
      },
    );

    if (response.statusCode != 200) {
      throw Exception('Failed to fetch home shelves: ${response.statusCode}');
    }

    final List<dynamic> jsonList = json.decode(response.body);
    return jsonList
        .map((json) => MusicHomeShelf.fromJson(json as Map<String, dynamic>))
        .toList();
  }

  String _getAccessToken() {
    // TODO: Get access token from auth context
    return '';
  }
}

final musicService = MusicService();
